### ISSUE TITLE
[Schema] Inline description and vendor extensions under allOf is not recognized

### ISSUE THREAD

**Issue by chenliu0831** [OpenAPITools/openapi-generator#1125]
##### Description
We have an use-case where `description` and vendor extensions are put inside `allOf` as inline schema. However those inlined fields went missing from the input to the generator as well as the final generated code.  See below:

```yaml
ModelA:
  allOf:
    - "$ref": "#/components/schemas/ModelBase"
    - type: object
      description: foo
      properties:
        value:
          type: integer
          description: TODO
      x-model-version: '1.0'
      x-model-module: model.base
  x-model-base: model.base.ModelBase@1.0
```

Debug output run with `-DdebugModels` which should be the same to any generators. Notice how top level `x-model-base` is getting picked up in vendorExtensions while the extensions are not. 

```json
"model" : {
    "parent" : "ModelBase",
    "parentSchema" : "ModelBase",
    "interfaces" : [ "ModelBase" ],
    "name" : "ModelA",
    "classname" : "ModelA",
    "classVarName" : "ModelA",
    "modelJson" : "{\n  \"allOf\" : [ {\n    \"$ref\" : \"#/components/schemas/ModelBase\"\n  }, {\n    \"type\" : \"object\",\n    \"properties\" : {\n      \"value\" : {\n        \"type\" : \"integer\",\n        \"description\" : \"TODO\",\n        \"format\" : \"int32\"\n      }\n    },\n    \"description\" : \"foo\",\n    \"x-model-version\" : \"1.0\",\n    \"x-model-module\" : \"model.base\"\n  } ],\n  \"x-model-base\" : \"model.base.ModelBase@1.0\"\n}",
    "classFilename" : "ModelA",
    "isAlias" : false,
    "isString" : false,
    "isInteger" : false,
    "vars" : [],
    "requiredVars" : [],
    "optionalVars" : [],
    "readOnlyVars" : [],
    "readWriteVars" : [],
    "allVars" : [],
    "parentVars" : [ ],
    "mandatory" : [ ],
    "allMandatory" : [ ],
    "imports" : [ "AttributeMetadata", "ModelBase" ],
    "hasVars" : true,
    "emptyVars" : false,
    "hasMoreModels" : false,
    "hasEnums" : false,
    "isEnum" : false,
    "hasRequired" : false,
    "hasOptional" : true,
    "isArrayModel" : false,
    "hasChildren" : false,
    "isMapModel" : false,
    "hasOnlyReadOnly" : false,
    "vendorExtensions" : {
      "x-model-base" : "model.base.ModelBase@1.0"
    }
  }
}
```

According to the [schema object spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#schemaObject), either way seems to be valid. Is this a bug, a feature by design or I misunderstand anything?

Thanks.

##### openapi-generator version

3.3.2, also happens with master.

**Comment by chenliu0831** [comment]
Could someone take a look? Thanks!

**Comment by wing328** [comment]
@chenliu0831 thanks for reporting the issue. I've filed https://github.com/OpenAPITools/openapi-generator/pull/1360 with better `allOf` support. I don't think it addresses the issue you reported but still would appreciate your feedback on the PR.

**Comment by chenliu0831** [comment]
@wing328 Great to see you working on this! A small suggestion for your PR #1360: could you add a few high level pointers to the module where we could look at the major change of those fixes/features. It's somewhat hard to follow for someone not familiar with the repository. Thanks!

**Comment by wing328** [comment]
@chenliu0831 #1360 has been merged into master. Please give it a try and let us know if you still see issues with `allOf`

**Comment by wing328** [comment]
> could you add a few high level pointers to the module where we could look at the major change of those fixes/features. 

Agree. I'll try to add more comments to the code so that new users can more easily follow.
